### PR TITLE
GA blocker review calling SDK

### PR DESCRIPTION
**Pull request by raosanat**
Hi Travis , regarding the 3 GA blockers (GA API review items  (Web view))

Swift friendly signatures for events.

Hide internal implementation in public header files.

Partial nullability type specifier implementation (constructors and completion blocks).

### CODE REVIEW

File: sdk/communication/AzureCommunicationCalling/AzureCommunicationCalling-Swift.h [hunk 1]
```diff
@@ -0,0 +1,1027 @@
...
+
+/// Local and Remote Video Stream types
+public enum MediaStreamType : Int {
+
+    
+    /// Video
+    case video = 0
+
+    /// Screen share
+    case screenSharing = 1
```
Comment (tjprescott) [inline sdk/communication/AzureCommunicationCalling/AzureCommunicationCalling-Swift.h:95]: Seems like most of the enum consistency issues are fixed except this. 1 and 2?

File: sdk/communication/AzureCommunicationCalling/AzureCommunicationCalling-Swift.h [hunk 2]
```diff
@@ -0,0 +1,1027 @@
...
+
+/**
+ * A set of methods that are called by ACSCallAgent in response to important events.
+ */
+public protocol CallAgentDelegate : NSObjectProtocol {
+
+    
+    optional func callAgent(_ callAgent: CallAgent!, didUpdateCalls args: CallsUpdatedEventArgs!)
+
+    optional func callAgent(_ callAgent: CallAgent!, didRecieveIncomingCall incomingcall: IncomingCall!)
```
Comment (tjprescott) [inline sdk/communication/AzureCommunicationCalling/AzureCommunicationCalling-Swift.h:263]: Looks good though `incomingcall` should probably be `incomingCall`.

**Review by tjprescott — CHANGES_REQUESTED** [review]
Still awaiting the changes to get rid of `!` everywhere, but the delegates look great! Thanks @raosanat. A couple minor comments otherwise.

File: sdk/communication/AzureCommunicationCalling/AzureCommunicationCalling-Swift.h (hunk 1 above)
Comment (raosanat) [reply sdk/communication/AzureCommunicationCalling/AzureCommunicationCalling-Swift.h:95]: yes those will be fixed in another PR, here we are only focussing on. the GA blockers.

File: sdk/communication/AzureCommunicationCalling/AzureCommunicationCalling-Swift.h [hunk 3]
```diff
@@ -0,0 +1,1027 @@
...
+    /// Video options when starting a call
+    open var videoOptions: VideoOptions?
+
+    
+    /// Audio options when starting a call
+    open var audioOptions: AudioOptions?
+
+    
+    // Class extension begins for StartCallOptions.
+    open var alternateCallerID: PhoneNumberIdentifier
```
Comment (tjprescott) [inline sdk/communication/AzureCommunicationCalling/AzureCommunicationCalling-Swift.h:462]: should be `alternateCalledId` per iOS guidelines.

File: sdk/communication/AzureCommunicationCalling/AzureCommunicationCalling-Swift.h [hunk 4]
```diff
@@ -0,0 +1,1027 @@
...
+
+    public init()
+
+    
+    /// Deallocates the memory occupied by this object.
+    open func dealloc()
+
+    
+    // Class extension begins for AddPhoneNumberOptions.
+    open var alternateCallerID: PhoneNumberIdentifier
```
Comment (tjprescott) [inline sdk/communication/AzureCommunicationCalling/AzureCommunicationCalling-Swift.h:479]: Same.

File: sdk/communication/AzureCommunicationCalling/AzureCommunicationCalling-Swift.h [hunk 5]
```diff
@@ -0,0 +1,1027 @@
...
+
+    public init()
+
+    
+    /// Deallocates the memory occupied by this object.
+    open func dealloc()
+
+    
+    /// Start an outgoing or accept incoming call muted (true) or un-muted(false)
+    open var muted: Bool
```
Comment (tjprescott) [inline sdk/communication/AzureCommunicationCalling/AzureCommunicationCalling-Swift.h:405]: Should probably be `isMuted` to align with other boolean properties in your SDK?  Perhaps `startMuted`?

File: sdk/communication/AzureCommunicationCalling/AzureCommunicationCalling-Swift.h (hunk 2 above)
Comment (raosanat) [reply sdk/communication/AzureCommunicationCalling/AzureCommunicationCalling-Swift.h]: Done will change it

File: sdk/communication/AzureCommunicationCalling/ACSRenderer-Swift.h [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+open class VideoStreamRenderer : NSObject {
+
+    
+    public init(localVideoStream: ACSLocalVideoStream) throws
+
+    public init(remoteVideoStream: ACSRemoteVideoStream) throws
+
+    open func createView() throws -> RendererView
+
+    open func createView(with options: ACSRenderingOptions?) throws -> RendererView
```
[the remark this answers is not in the document]
Comment (tjprescott) [reply sdk/communication/AzureCommunicationCalling/ACSRenderer-Swift.h:25]: Needs to be fixed before GA

[the remark this answers is not in the document]
Comment (raosanat) [reply sdk/communication/AzureCommunicationCalling/ACSRenderer-Swift.h:25]: yes fixing it now

File: sdk/communication/AzureCommunicationCalling/AzureCommunicationCalling-Swift.h (hunk 2 above)
Comment (raosanat) [reply sdk/communication/AzureCommunicationCalling/AzureCommunicationCalling-Swift.h]: Done changed

File: sdk/communication/AzureCommunicationCalling/ACSRenderer-Swift.h [hunk 7]
```diff
@@ -0,0 +1,33 @@
+//+-----------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+//
+// Module name : ACSRenderer.h
+//
+//------------------------------------------------------------------------------
+
+public protocol RendererDelegate : NSObjectProtocol {
+
+    func videoStreamRenderer(didFailToStartRenderer renderer: VideoStreamRenderer)
```
[the remark this answers is not in the document]
Comment (raosanat) [reply sdk/communication/AzureCommunicationCalling/ACSRenderer-Swift.h:10]: Done updated.

File: sdk/communication/AzureCommunicationCalling/ACSRenderer-Swift.h (hunk 6 above)
[the remark this answers is not in the document]
Comment (raosanat) [reply sdk/communication/AzureCommunicationCalling/ACSRenderer-Swift.h:25]: Done

File: sdk/communication/AzureCommunicationCalling/AzureCommunicationCalling-Swift.h (hunk 3 above)
Comment (tjprescott) [reply sdk/communication/AzureCommunicationCalling/AzureCommunicationCalling-Swift.h:462]: Needs to be fixed by GA

**Review by tjprescott — APPROVED** [review]
There a few outstanding issues that need to be taken care of, but otherwise this is good to go for GA.

**Comment by tjprescott** [comment]
Review is approved. Closing.

**Pull request closed without merging**